src/sass/modules/_m-schemaform.scss

```diff
@@ -77,3 +81,36 @@
 .schemaform-array-row {
   outline: none;
 }
+
+.schemaform-block-header {
+  margin-bottom: 0.5em;
+  > legend, > h5, > p {
+    padding-bottom: 0 !important;
+    margin-bottom: 0 !important;
+  }
+  > p {
+    margin-top: 0.5em;
+  }
+}
+
+// This is the worst
```

Comment:
```css
.schemaform-block-header + .schemaform-first-field,
.schemaform-first-field--root .schemaform-first-field {
  label, .usa-input-error {
    margin-top: 0;
  }
}
```

Seems solid, but I'm not familiar with all the uses of this.

If we go this way, we won't even need `.schemaform-block-header--titled` up in `ObjectField` (and `AddressField`, probably).